exercises/static/exercises/human_detection_newmanager/python_template/ros2_humble/HAL.py

```diff
@@ -0,0 +1,49 @@
+import rclpy
+from rclpy.node import Node
+from sensor_msgs.msg import Image
+from cv_bridge import CvBridge
+import threading
+import cv2
+
+current_frame = None  # Global variable to store the frame
+
+class WebcamSubscriber(Node):
```

Comment:
Hi, Instead of creating a local subscriber you can use a HAL interface called CameraNode see an example here:
https://github.com/JdeRobot/RoboticsAcademy/blob/532c8eeee63536ebfb6a0c532c6d4b251c784edd/exercises/static/exercises/rescue_people_newmanager/python_template/ros2_humble/HAL.py#L24

Reply:
Trying to use the CameraNode HAL interface was unsuccessful in capturing frames due to the physical camera setup. Consequently, I had to switch to a separate subscriber code.